hdfs-dataset-sink/src/test/java/org/springframework/cloud/stream/module/dataset/sink/DatasetSinkConfigurationTests.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.dataset.sink;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.data.hadoop.fs.FsShell;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Thomas Risberg
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = DatasetSinkApplication.class)
+@WebIntegrationTest({"server.port:0",
+		"spring.hadoop.fsUri=file:///",
+		"directory=${java.io.tmpdir}/dataset",
```

Comment:
I see.
Wouldn't it be better to move the ${java.io.tmpdir} part to
`spring.hadoop.fsUri`
though (`spring.hadoop.fsUri=file://${java.io.tmpdir}`) and drop it from
`directory` ?

On Wed, Nov 4, 2015 at 9:25 PM, Thomas Risberg <notifications@github.com>
wrote:

> In
> hdfs-dataset-sink/src/test/java/org/springframework/cloud/stream/module/dataset/sink/DatasetSinkConfigurationTests.java
> <https://github.com/spring-cloud/spring-cloud-stream-modules/pull/58#discussion_r43934442>
> :
>
> > +import org.springframework.test.annotation.DirtiesContext;
> > +import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
> > +
> > +import java.io.File;
> > +import java.io.IOException;
> > +
> > +import static org.junit.Assert.assertTrue;
> > +
> > +/**
> > + * @author Thomas Risberg
> > + */
> > +@RunWith(SpringJUnit4ClassRunner.class)
> > +@SpringApplicationConfiguration(classes = DatasetSinkApplication.class)
> > +@WebIntegrationTest({"server.port:0",
> > +		"spring.hadoop.fsUri=file:///",
> > +		"directory=${java.io.tmpdir}/dataset",
>
> it is, but with the fsUri set to file:/// it actually writes to the local
> file system through hdfs, so using a tmp directory made sense
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spring-cloud/spring-cloud-stream-modules/pull/58/files#r43934442>
> .
>